src/AutorestSwift/Models/Protocols.swift

```diff
@@ -9,11 +9,34 @@ import Foundation
 
 /// Custom extensible metadata for individual protocols (ie, HTTP, etc)
 public struct Protocols: Codable {
-    public let http: Protocol?
+    public let http: ProtocolInterface?
 
-    public let amqp: Protocol?
+    public let amqp: ProtocolInterface?
 
-    public let mqtt: Protocol?
+    public let mqtt: ProtocolInterface?
 
-    public let jsonrpc: Protocol?
+    public let jsonrpc: ProtocolInterface?
+
+    enum CodingKeys: String, CodingKey {
+        case http, amqp, mqtt, jsonrpc
+    }
+
+    // MARK: Codable
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        http = (try? container.decode(HttpParameter.self, forKey: .http)) ??
+            (try? container.decode(HttpResponse.self, forKey: .http)) ??
+            (try? container.decode(HttpModel.self, forKey: .http))
```

Comment:
You have to be careful the way you order these. They need to be from most specific/complex to least.

Reply:
you need to add a case 'HttpWithBodyRequest' to decode Http

Reply:
Done.